qualang_tools/config/waveform_tools.py

```diff
@@ -83,3 +87,90 @@ def drag_cosine_pulse_waveforms(amplitude, length, alpha, delta, detuning=0):
     I_wf = z.real.tolist()  # The `I` component is the real part of the waveform
     Q_wf = z.imag.tolist()  # The `Q` component is the imaginary part of the waveform
     return I_wf, Q_wf
+
+
+def flattop_gaussian_waveform(
+    amplitude, flat_length, rise_fall_length, return_part="all"
+):
+    """
+    Returns a flat top Gaussian waveform. This is a square pulse with a rise and fall of a Gaussian with the given
+    sigma. It is possible to only get the rising or falling parts, which allows scanning the flat part length from QUA.
+    The length of the pulse will be the `flat_length + 2 * rise_fall_length`.
+
+    :param float amplitude: The amplitude in volts.
+    :param int flat_length: The flat part length in ns.
+    :param float rise_fall_length: The rise and fall times in ns. The Gaussian sigma is given by the
+        `rise_fall_length / 4`.
+    :param str return_part: When set to 'all', returns the complete waveform. Default is 'all'. When set to 'rise',
+    returns only the rising part. When set to 'fall', returns only the falling part. This is useful for separating
+    the three parts which allows scanning the duration of the  flat part is to scanned from QUA
+    :return: Returns the waveform as a list
+    """
+    gauss_wave = amplitude * gaussian(2 * rise_fall_length, rise_fall_length / 4)
+
+    rise_part = gauss_wave[:rise_fall_length]
+    rise_part = rise_part.tolist()
+    if return_part == "all":
+        return rise_part + [amplitude] * flat_length + rise_part[::-1]
+    elif return_part == "rise":
+        return rise_part
+    elif return_part == "fall":
+        return rise_part[::-1]
+    else:
+        raise Exception("'return_part' must be either 'all', 'rise' or 'fall'")
+
+
+def flattop_cosine_waveform(
+    amplitude, flat_length, rise_fall_length, return_part="all"
+):
+    """
+    Returns a flat top cosine waveform. This is a square pulse with a rise and fall with cosine shape with the given
+    sigma. It is possible to only get the rising or falling parts, which allows scanning the flat part length from QUA.
+    The length of the pulse will be the `flat_length + 2 * rise_fall_length`.
+
+    :param float amplitude: The amplitude in volts.
+    :param int flat_length: The flat part length in ns.
+    :param float rise_fall_length: The rise and fall times in ns, taken as a number of points of a cosine between 0
+        and pi.
```

Comment:
I would say "taken as the time for a cosine to go from 0 to 1 (pi phase-shift) and conversely."